### DISCUSSION TITLE
DAI 2024-2025 - Practical work 2 - Mathéo Lopez, Emily Baquerizo, Kimberly Beyeler

### DISCUSSION THREAD

**Discussion by MathLopez** [heig-vd-dai-course/heig-vd-dai-course#492] [Show and tell]
## Project Description
We aim to create a digital implementation of the classic game *Mastermind* with a client-server architecture to enable remote gameplay. In this game, one player (the "server") generates a hidden code, while another player (the "client") attempts to guess the code within a limited number of turns. Each turn, the server provides feedback on the client’s guess, indicating the number of correct color and position matches (exact matches) and the number of correct colors in incorrect positions (partial matches). The client uses this feedback to refine its guesses progressively.

### Server
- Initializes the game by generating a random code.
- Processes each guess received from the client, compares it to the hidden code, and provides feedback on correct colors and positions.
- Enforces game rules, such as turn limits, and sends relevant game state updates to the client.
- Keeps track of scores, stores them in a file, and displays the leaderboard at the end of every game.

### Client
- Allows the player to input their guesses.
- Displays feedback from the server to help the player make educated guesses.
- Uses `picocli` to handle command-line input and display.

## Technologies
- **Networking Protocols:** TCP for communication between the client and the server.
- **Java I/O:** Used for score-keeping in files on the server.

**Comment by ludelafo** [comment]
Looks good to us, just don't forget your server must handle multiple games (each client has its own game).

**Comment by AEBaq** [comment]
https://github.com/HEIG-VD-BA5/DAI

**Comment by AEBaq** [comment]
Bonsoir, navrée on avait juste lu la présentation du cours où il était uniquement noté de répondre avec le repo, nous n'avions pas vu qu'il fallait répondre également avec le hash du commit... Vraiment navrée, voici le hash du dernier commit  : 6e6a6583f8e70549600fce2277a08437295715e5
